src/connectivity.jl

```diff
@@ -536,17 +536,27 @@ end
 """
     isgraphical(degs)
 
-Return true if the degree sequence `degs` is graphical, according to
-[Erdös-Gallai condition](http://mathworld.wolfram.com/GraphicSequence.html).
+Return true if the degree sequence `degs` is graphical
+According to Erdös-Gallai theorem, a degree sequence ``\\{d_1, ...,d_n\\}`` (sorted in descending order) is graphic iff the sum of vertex degrees is even and the sequence obeys the property -
+```math
+\\sum_{i=1}^{r} d_i \\leq r(r-1) + \\sum_{i=r+1}^n min(r,d_i)
+```
+for each integer r <= n-1
 
 ### Performance
-    Time complexity: ``\\mathcal{O}(|degs|^2)``
+Time complexity: ``\\mathcal{O}(|degs|)``.
 """
 function isgraphical(degs::Vector{Int})
     iseven(sum(degs)) || return false
-    n = length(degs)
+    sorted_degs = sort(degs, rev = true)
+    n = length(sorted_degs)
+    cur_sum = 0
+    mindeg = (i->min(i, sorted_degs[i])).(sorted_degs)
```

Comment:
I think you will get in trouble here, if an element of `sorted_degs` is not in the range `1:length(sorted_degs)`.